### DISCUSSION TITLE
Replica Cluster Source Node Selection and Read Operations Across Regions

### DISCUSSION THREAD

**Discussion by aryklein** [cloudnative-pg/cloudnative-pg#5322] [Q&A] [answered]
In the case of using a replica cluster (Deployments across Kubernetes clusters) with **pg_basebackup** (using streaming replication), I have a question regarding the source node. Can I point to any node in the cluster as the source, or does it have to be the primary node? The official documentation diagrams and examples seem to indicate that the source should always be the primary node. However, in my specific scenario, all my nodes are registered in Consul under the same name, so I would like to point to any of them.

Additionally, can I use the replica cluster not just as a fallback but also for read operations to reduce latency for queries executed in another region? Thank you very much

**Comment by aryklein** [comment]
btw, I wasn't able to find the `replication_slot` for the replica cluster, but I do see the replication slots for the replicas in the same cluster.

**Comment by gbartolini** [comment, marked as answer]
Hi @aryklein,

Thank you for using CloudNativePG.

You should be able to point a replica cluster to any service in the origin, provided you're aware of the implications. For instance, you could use the `-r` service to point to any instance, or `-ro` if you’re targeting a replica.

You can definitely use the replica cluster for read operations—just be sure you’re familiar with the limitations of the hot standby feature.

I hope this helps!

Best regards,
Gabriele

**Reply by aryklein** [reply]
Thanks @gbartolini for your answer and for your work on this amazing operator.

Is there no replication slot for replica clusters?

Best,
Ary
